apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.js

```diff
@@ -7,325 +7,272 @@ const sqldb = require('@prairielearn/postgres');
+          // Because of the redirect, if the job sequence ends up failing to save,
+          // then the corresponding draft will be lost (all drafts are soft-deleted
+          // from the database on readDraftEdit).
+          debug('Job sequence is still running - redirect to status page');
+          res.redirect(`${res.locals.urlPrefix}/jobSequence/${fileEdit.jobSequenceId}`);
+          return;
         }
-
-        const data = await editorUtil.getErrorsAndWarningsForFilePath(
-          res.locals.course.id,
-          relPath,
-        );
-        const ansiUp = new AnsiUp();
-        fileEdit.sync_errors = data.errors;
-        fileEdit.sync_errors_ansified = ansiUp.ansi_to_html(fileEdit.sync_errors);
-        fileEdit.sync_warnings = data.warnings;
-        fileEdit.sync_warnings_ansified = ansiUp.ansi_to_html(fileEdit.sync_warnings);
-      },
-    ],
-    (err) => {
-      if (ERR(err, next)) return;
-      if ('jobSequence' in fileEdit && fileEdit.jobSequence.status === 'Running') {
-        debug('Job sequence is still running - redirect to status page');
-        res.redirect(`${res.locals.urlPrefix}/jobSequence/${fileEdit.jobSequenceId}`);
-        return;
-      }
-
-      fileEdit.alertChoice = false;
-
-      if ('editID' in fileEdit) {
-        // There is a recently saved draft ...
-        fileEdit.alertResults = true;
-        if (!fileEdit.didSave && fileEdit.editHash !== fileEdit.diskHash) {
-          // ...that was not written to disk and that differs from what is on disk.
-          fileEdit.alertChoice = true;
-          fileEdit.hasSameHash = fileEdit.origHash === fileEdit.diskHash;
+  
+        fileEdit.alertChoice = false;
+        fileEdit.didSave = false;
+        fileEdit.didSync = false;
+  
+        if ('jobSequence' in fileEdit) {
+          // No draft is older than 24 hours, so it is safe to assume that no
+          // job sequence is legacy... but, just in case, we will check and log
+          // a warning if we find one. We will treat the corresponding draft as
+          // if it was neither saved nor synced.
+          if (fileEdit.jobSequence.legacy) {
+            debug('Found a legacy job sequence');
+            logger.warn(`Found a legacy job sequence (id=${fileEdit.jobSequenceId}) ` +
+                        `in a file edit (id=${fileEdit.editID})`);
+          } else {
+            const job = fileEdit.jobSequence.jobs[0];
+  
+            debug(`Found a job sequence: ` +
+              `syncAttempted=${job.data.syncAttempted}, ` +
+              `syncSucceeded=${job.data.syncSucceeded}`);
+            
+            // We check for the presence of a `syncAttempted` key to know if
+            // we attempted a sync (if it exists, its value will be true). If
+            // a sync was attempted, then the edit must have been saved (i.e,
+            // written to disk in the case of no git, or written to disk and
+            // then pushed in the case of git).
```

Comment:
I'm not actually sure if this is true. As far as I can tell, we set `syncAttempted` before as actually attempt to add/commit/push the changes.

Reply:
Fixed.